skorch/classifier.py

```diff
@@ -188,3 +189,160 @@ def predict(self, X):
             y_preds.append(to_numpy(yp.max(-1)[-1]))
         y_pred = np.concatenate(y_preds, 0)
         return y_pred
+
+
+neural_net_binary_clf_doc_start = """NeuralNet for binary classification tasks
+
+    Use this specifically if you have a binary classification task,
+    with input data X and target y. y must be 1d.
+
+"""
+
+neural_net_binary_clf_criterion_text = """
+
+    criterion : torch criterion (class, default=torch.nn.BCEWithLogitsLoss)
+      Binary cross entropy loss with logits. Note that the module should return
+      the logit of probabilities with shape (batch_size, )."""
+
+
+def get_neural_net_binary_clf_doc(doc):
+    doc = neural_net_binary_clf_doc_start + " " + doc.split("\n ", 4)[-1]
+    pattern = re.compile(r'(\n\s+)(criterion .*\n)(\s.+){1,99}')
+    start, end = pattern.search(doc).span()
+    doc = doc[:start] + neural_net_binary_clf_criterion_text + doc[end:]
+    return doc
+
+
+# pylint: disable=missing-docstring
+class NeuralNetBinaryClassifier(NeuralNet):
+    __doc__ = get_neural_net_binary_clf_doc(NeuralNet.__doc__)
+
+    def __init__(
+            self,
+            module,
+            criterion=torch.nn.BCEWithLogitsLoss,
+            train_split=CVSplit(5, stratified=True),
+            *args,
+            **kwargs
+    ):
+        super().__init__(
+            module,
+            criterion=criterion,
+            train_split=train_split,
+            *args,
+            **kwargs
+        )
+
+    @property
+    def _default_callbacks(self):
+        return [
+            ('epoch_timer', EpochTimer()),
+            ('train_loss', BatchScoring(
+                train_loss_score,
+                name='train_loss',
+                on_train=True,
+                target_extractor=noop,
+            )),
+            ('valid_loss', BatchScoring(
+                valid_loss_score,
+                name='valid_loss',
+                target_extractor=noop,
+            )),
+            ('valid_acc', EpochScoring(
+                'accuracy',
+                name='valid_acc',
+                lower_is_better=False,
+            )),
+            ('print_log', PrintLog()),
+        ]
+
+    # pylint: disable=signature-differs
+    def check_data(self, X, y):
+        super().check_data(X, y)
+        if get_dim(y) != 1:
+            raise ValueError("The target data should be 1-dimensional.")
+
+    # pylint: disable=signature-differs
+    def fit(self, X, y, **fit_params):
+        """See ``NeuralNet.fit``.
+
+        In contrast to ``NeuralNet.fit``, ``y`` is non-optional to
+        avoid mistakenly forgetting about ``y``. However, ``y`` can be
+        set to ``None`` in case it is derived dynamically from
+        ``X``.
+
+        """
+        # pylint: disable=useless-super-delegation
+        # this is actually a pylint bug:
+        # https://github.com/PyCQA/pylint/issues/1085
+        return super().fit(X, y, **fit_params)
+
+    def predict(self, X, threshold=0.5):
```

Comment:
I think this needs to be a class attribute since passing kwargs in predict is enerally unsupported by sklearn. This would not work in a pipeline for example.